dcap-ql/src/quote.rs

```diff
@@ -229,6 +301,45 @@ impl<'a> Qe3CertData<'a> for Qe3CertDataPpid<'a> {
     }
 }
 
+impl<'a> Qe3CertData<'a> for RawQe3CertData<'a>{
+    fn parse(_type_: CertificationDataType, data: &Cow<'a, [u8]>) -> Result<Self> {
+        Ok(data.clone())
+    }
+}
+
+fn find_pos_mark(data: &[u8], mark: &[u8]) -> Option<usize> {
+    if data.get(0..mark.len()) == Some(mark) {
+        Some(0)
+    } else if data.len() >= mark.len() && data.len() > 0 {
+        find_pos_mark(data.get(1..)?, mark).and_then(|idx| Some(idx + 1))
+    } else {
+        None
+    }
+}
+
+impl<'a> Qe3CertData<'a> for Qe3CertDataPckCertChain<'a> {
+    fn parse(type_: CertificationDataType, data: &Cow<'a, [u8]>) -> Result<Self>{
+        if type_ != CertificationDataType::PckCertificateChain {
+            bail!("Invalid certification data type: {:?}", type_);
+        }
+
+        let mut data = data.clone();
+
+        let mut certs = vec![];
+        let mark = "-----END CERTIFICATE-----".as_bytes();
+        while let Some(pos) = find_pos_mark(&data, mark) {
```

Comment:
Please use `str::find` instead of `find_pos_mark`.

You can convert to Cow<str> like so:
```
let data = match data {
    Cow::Borrowed(s) =>
        str::from_utf8(s)
            .map(Cow::Borrowed)
            .map_err(|e| format_err!("Invalid certificate format: {}", e))?,
    Cow::Owned(s) =>
        str::from_utf8(s)
            .map(Cow::Owned)
            .map_err(|e| format_err!("Invalid certificate format: {}", e))?,
};
```